From: Deepak K <deepak.k@ti.com>
Date: Fri, 13 May 2011 23:28:00 +0000 (-0500)
Subject: OMAP: UART : Fixes the boot failure for lower baud rates
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=131d23ac0b0eece4de960689ca7394eaae191489

OMAP: UART : Fixes the boot failure for lower baud rates

This patch fixes the boot up of LInux at lower baud rates. Due to
timing issues between the platform device and probe of the driver
getting called, the Inactivity timers were getting called even
when the Uart Port was not initialized. This results in crash
at boot up. A Null check would fix this issue as this would allow
the Timer to come up and exit gracefully.

Change-Id: I62499e51d6f420af0517d2e2d3bc0efb8cebb518
Signed-off-by: Deepak K <deepak.k@ti.com>
---

Index: kernel/drivers/serial/omap-serial.c
===================================================================
--- kernel.orig/drivers/serial/omap-serial.c	2011-05-21 00:43:57.309639169 +0900
+++ kernel/drivers/serial/omap-serial.c	2011-05-21 00:45:06.040889433 +0900
@@ -1626,6 +1626,15 @@
 	struct circ_buf *xmit;
 	unsigned int status;
 
+	/* Though when UART's initialised this can never happen,
+	 * but during initialisation, it can happen the "ui"
+	 * structure is not initialized and the timer kicks
+	 * in. This would result in a NULL value, resulting
+	 * in crash.
+	 */
+	if (up == NULL)
+		return 0;
+
 	/* Check for recent driver activity. If time delta from now
 	 * to last activty < "uart idle timeout" second keep clocks on.
 	 */
